libs/wingc/src/lsp/sync.rs

```diff
@@ -106,7 +106,12 @@ fn partial_compile(source_file: &str, text: &[u8]) -> FileData {
 	let mut capture_visitor = CaptureVisitor::new();
 	capture_visitor.visit_scope(&scope);
 
-	let diagnostics = vec![parse_diag, type_diag, capture_visitor.diagnostics].concat();
+	let mut diagnostics = Diagnostics::new();
+	for diags in [parse_diag, type_diag, capture_visitor.diagnostics].iter() {
+		for diag in diags.iter() {
+			diagnostics.insert(diag.clone());
+		}
+	}
```

Comment:
Feels like there's probably a way to do this without the extra clones...

Reply:
Played around with this a bit. I had a version that uses `IndexSet.union` but that was clunky and broke when we added the capture visitor (after a merge from main). Maybe worth andding a `concat` like thing `IndexSet` and PRing it to them..
I'm open to any improvements, but eventually just wanted to get it done...